agent/exec/container/adapter.go

```diff
@@ -61,15 +61,21 @@ func (c *containerAdapter) pullImage(ctx context.Context) error {
 
 	dec := json.NewDecoder(rc)
 	m := map[string]interface{}{}
+	spamLimiter := time.NewTicker(time.Millisecond * 500).C
```

Comment:
The ticker needs to be stopped with `Stop` after it's done being used. You could defer that here.